Trippy/Utils/ParseUtils.m

```diff
@@ -100,51 +102,57 @@ + (Location *)locationFromPFObj:(PFObject *)obj {
     return [[Location alloc] initWithParams:obj[@"title"] snippet:obj[@"snippet"] latitude:coord.latitude longitude:coord.longitude user:user.username placeId:obj[@"placeId"] parseObjectId:obj.objectId];
 }
 
-+ (PFObject *)newPFObjWithCollection:(LocationCollection *)collection {
-    PFObject *obj = [PFObject objectWithClassName:@"Collection"];
-    obj[@"title"] = collection.title;
-    obj[@"snippet"] = collection.snippet;
-    obj[@"createdBy"] = [PFUser currentUser];
-    PFRelation *relation = [obj relationForKey:@"locations"];
-    for(Location *loc in collection.locations) {
-        PFObject *newObj = [self oldPFObjFromLocation:loc];
-        [relation addObject: newObj];
++ (PFObject *)pfObjFromCollection:(LocationCollection *)collection {
+    if (collection.parseObjectId != nil) {
+        PFQuery *query = [PFQuery queryWithClassName:@"Collection"];
+        [query includeKeys:[ParseUtils getCollectionKeys]];
+        return [query getObjectWithId:collection.parseObjectId];
+    } else {
+        PFObject *obj = [PFObject objectWithClassName:@"Collection"];
+        obj[@"title"] = collection.title;
+        obj[@"snippet"] = collection.snippet;
+        obj[@"createdBy"] = [PFUser currentUser];
+        PFRelation *relation = [obj relationForKey:@"locations"];
+        for(Location *loc in collection.locations) {
+            PFObject *newObj = [self pfObjFromLocation:loc];
+            [relation addObject: newObj];
+        }
+        return obj;
```

Comment:
Do you need to set the `parseObjectId` of collection here, or do you do it somewhere else in the stack? You might consider moving it here to keep it a little more tidy.